src/browser/page-switch.js

```diff
@@ -31,17 +29,26 @@ define((require, exports, moudle) => {
         }
       }
     }, [
-      DOM.span({
+      DOM.div({
         key: 'thumbnail',
         className: 'tab-thumbnail',
-        style: {backgroundImage: readThumbnailURI(webViewerCursor.get('location'))},
-      }),
+      }, [
+        DOM.img({
+          key: 'image',
+          className: 'image',
+          src: thumbnail,
+          alt: '',
+          onLoad: event => URL.revokeObjectURL(event.target.src),
```

Comment:
That's suspicious. This will revoke the URL you just loaded and next gc will discard your image.